app/javascript/packages/validated-field/validated-field.tsx

```diff
@@ -80,14 +80,12 @@ function ValidatedField(
 ) {
   const fieldRef = useRef<ValidatedFieldElement>();
   const instanceId = useInstanceId();
-  useImperativeHandle(forwardedRef, () => ({
-    reportValidity: () => fieldRef.current?.input?.reportValidity(),
-  }));
+  useImperativeHandle(forwardedRef, () => fieldRef.current?.input);
   useEffect(() => {
     if (fieldRef.current && fieldRef.current.input) {
       const { input } = fieldRef.current;
       input.checkValidity = () => {
-        let nextError: string = '';
+        let nextError: string = (input.validity.customError && input.validationMessage) || '';
```

Comment:
This looks like it's probably a good change, but I think we should probably set it after the `validate` call (if not set in the `catch`) and add a test to prevent regression.

Reply:
I will look into this more, thanks. 👍🏻

Reply:
@tomas-nava since you're picking this story up - FYI I think this line is the only one remaining that I think should be addressed before approval.

Reply:
fixed in a8192899099876c0343f878181c8408aaf194d2f